---
title: "220801_mouse_diarrhea_reproducible"
author: "Bryan Merrill"
date: '2022-08-01'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(data.table)
library(tidytable)
library(ggplot2)
library(ape)
library(plotly)
library(pheatmap)
library(readxl)
library(ggsignif)
library(phyloseq)
library(ggbeeswarm)
library(ggheatmap)
library(seriation)
library(phyloseq)
library(cowplot)
library(rstatix)
library(RColorBrewer)
```

# Reading in files
```{r}
ps <- readRDS("mouse_diarrhea_phyloseq_coverage.rds")
ps_rel <- ps %>% 
  filter_taxa(function(x){sum(x>0)>3}, prune=T) %>% # removes 58 taxa that are present in <= 3 samples
  transform_sample_counts(function(OTU) OTU/sum(OTU) )

sample_order <- sample_data(ps_rel) %>% data.frame() %>% rownames_to_column("SampleName") %>% arrange(index2) %>% pull(SampleName)


md_metadata <- sample_data(ps_rel) %>% 
            rownames_to_column("SampleName") %>% data.frame() %>% 
            select(SampleName, everything())

write_tsv(md_metadata, 
          "tables/table_s1_sample_metadata.tsv")
```


# FIGURE 1
## Phylum relative abundance plot
```{r}
ps_phylum <- ps_rel %>% 
  tax_glom(taxrank="Phylum")

phylum_otudata <- otu_table(ps_phylum) %>% 
  data.frame() %>% 
  rownames_to_column("p") %>% 
  left_join(tax_table(ps_phylum) %>% data.frame() %>% 
              rownames_to_column("p") %>% 
              select("p","Phylum")) %>% 
  select(Phylum, everything(), -p) %>% 
  pivot_longer(2:27, names_to = "SampleName", values_to="rel_abundance") %>% 
  left_join(sample_data(ps_phylum) %>% data.frame() %>% 
              rownames_to_column("SampleName") %>% 
              select(SampleName, index2, Experiment, diarrhea)) %>% 
  arrange(index2, Phylum, Experiment) %>% 
  mutate(index2 = factor(index2, levels=seq(1:26)),
         Phylum = gsub("p__","",Phylum))

ggplot(phylum_otudata, aes(x=index2, y=rel_abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  facet_grid(~Experiment, scales = "free_x") + 
  theme_cowplot() + 
  theme(axis.ticks.x=element_blank(),
        axis.text.x=element_blank(),
        strip.text.x=element_blank(), axis.line.x=element_blank(),  
          axis.line.y=element_blank(), axis.title.x=element_blank()) + 
  ylab("Relative abundance") + 
  scale_y_continuous(n.breaks = 2)

ggsave("figures/Fig_1_relAb_by_mouse.pdf")
```
Phylum OTU data (mean)
```{r}
phylum_otudata_mean <- phylum_otudata %>% 
  summarize.(mean_rel_ab = mean(rel_abundance), .by=c("Phylum","Experiment"))

ggplot(phylum_otudata_mean, aes(x=Experiment, y=mean_rel_ab, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  facet_grid(~Experiment, scales = "free_x") + 
  theme_cowplot() + 
  theme(axis.ticks.x=element_blank(),
        axis.text.x=element_blank(),
        strip.text.x=element_blank(), axis.line.x=element_blank(),  
          axis.line.y=element_blank(), axis.title.x=element_blank()) + 
  ylab("Relative abundance") + 
  scale_y_continuous(n.breaks = 2)
```
## Stat signif phylum
```{r}
wilcox_p_phylum <- phylum_otudata %>%
  # filter(Phylum=="Bacteroidota" | Phylum == "Firmicutes_A" | Phylum == "Firmicutes" | Phylum == "Actinobacteriota") %>%
  # filter(Experiment == "Control 2" | Experiment == "Co-housed w/ diarrhea") %>%
  group_by(Phylum) %>%
  wilcox_test(rel_abundance~diarrhea) %>%
  adjust_pvalue()
  
  
anova_p_phylum <- phylum_otudata %>% 
  filter(Experiment != "Control 2") %>%
  group_by(Phylum) %>% 
  anova_test(rel_abundance~Experiment) %>% 
  adjust_pvalue()
```

## Stats
```{r}
tax_table_out <- tax_table(ps_rel) %>% data.frame() %>% rownames_to_column("genome")

phylum_taxmap <- rbind(tax_table_out %>% select(3) %>% mutate(rank_level="p") %>% distinct() %>% mutate(tax_rank=Phylum),
                        tax_table_out %>% select(3:4) %>% mutate(rank_level="c") %>% distinct() %>% dplyr::rename(tax_rank=Class),
                        tax_table_out %>% select(3,5) %>% mutate(rank_level="o") %>% distinct() %>% dplyr::rename(tax_rank=Order),
                        tax_table_out %>% select(3,6) %>% mutate(rank_level="f") %>% distinct() %>% dplyr::rename(tax_rank=Family),
                        tax_table_out %>% select(3,7) %>% mutate(rank_level="g") %>% distinct() %>% dplyr::rename(tax_rank=Genus),
                        tax_table_out %>% select(3,8) %>% mutate(rank_level="s") %>% distinct() %>% dplyr::rename(tax_rank=Species)) %>% 
  mutate(Phylum=gsub("p__","",Phylum))

taxa_abundance_long <- otu_table(ps_rel) %>% data.frame() %>% 
  rownames_to_column("genome") %>% 
  pivot_longer(2:27, names_to="SampleName", values_to="rel_abundance") %>% 
  left_join(tax_table_out)

make_taxa_summ_table <- function(x, df){
  out <- summarize.(df, rel_abundance=sum(rel_abundance), .by=c("SampleName",x)) %>% 
    dplyr::rename(tax_rank=x) %>% 
    mutate(rank_level=x)
  return(out)
}

taxa_level_summary <- rbind(make_taxa_summ_table("Phylum", taxa_abundance_long),
                              make_taxa_summ_table("Class", taxa_abundance_long),
                              make_taxa_summ_table("Order", taxa_abundance_long),
                              make_taxa_summ_table("Family", taxa_abundance_long),
                              make_taxa_summ_table("Genus", taxa_abundance_long),
                              make_taxa_summ_table("Species", taxa_abundance_long)) %>% 
  left_join.(sample_data(ps_rel) %>% data.frame() %>% 
               rownames_to_column("SampleName") %>% select.(SampleName, diarrhea, exp_group, Experiment))


taxa_level_stats <- taxa_level_summary %>%
  group_by(diarrhea, tax_rank) %>%
  summarize.(mean=mean(rel_abundance), .by=c(diarrhea, tax_rank, rank_level)) %>%
  pivot_wider.(names_from=diarrhea, values_from=mean) %>%
  mutate.(rel_ab_change=(no/yes))

wilcox_effsize <- taxa_level_summary %>% 
  group_by(tax_rank) %>% 
  wilcox_effsize(data=., rel_abundance~diarrhea)

wilcox_test_results <- taxa_level_summary %>% 
  group_by(tax_rank) %>% 
  wilcox_test(data=., rel_abundance~diarrhea) %>% 
  adjust_pvalue() %>% 
  separate.(tax_rank, sep="__", into="rank_level", remove=F) %>% 
  left_join.(taxa_level_summary %>% summarize.(y.position=max(rel_abundance)+0.5*sd(rel_abundance), .by=tax_rank)) %>% 
  left_join.(wilcox_effsize %>% select.(tax_rank, effsize, magnitude)) %>% 
  left_join.(taxa_level_stats %>% select.(tax_rank, no, yes, rel_ab_change)) %>% 
  mutate.(effsize_trans=ifelse(no>yes, -effsize, effsize),
         log.p.adj=-log10(p.adj)) %>% 
  left_join.(phylum_taxmap) %>% 
  mutate(rank_level_full=case_when(
           rank_level == "p" ~ "Phylum",
           rank_level == "c" ~ "Class",
           rank_level == "o" ~ "Order",
           rank_level == "f" ~ "Family",
           rank_level == "g" ~ "Genus",
           rank_level == "s" ~ "Species",
         ),
         rank_level=factor(rank_level, levels=c("p","c","o","f","g","s")),
         rank_level_full=factor(rank_level_full, levels=c("Phylum","Class","Order","Family","Genus","Species"))) %>% 
  arrange(rank_level_full)

write_tsv(wilcox_test_results, "tables/wilcox_test_results.tsv")

volc_plot <- ggplot(wilcox_test_results, aes(x=effsize_trans, y=log.p.adj, color=Phylum)) +
  facet_wrap(~rank_level_full, scales="free_x") + 
  geom_hline(yintercept = 1.30103) + 
  geom_jitter(width=0.2, height=0.2, shape=1, size=2) + 
  ylab("-log10(p.adj)") + xlab("effect size") + 
  # ggtitle("Effect size vs significance for taxonomic ranks") + 
  scale_x_continuous(breaks=c(-1,0,1), limits=c(-1.2,1.2)) + 
  theme_cowplot()
volc_plot
ggsave("figures/Fig_S1_volc_plot_all.pdf", plot = volc_plot, width=8, height=5)
```

```{r}
species_p <- ggplot(wilcox_test_results %>% filter(rank_level=="s"), aes(x=effsize_trans, y=log.p.adj, label=tax_rank, color=Phylum)) + 
  geom_jitter(width=0.25, height=0.25, shape=1, size=2) + 
  geom_hline(yintercept = 1.30103) + 
  facet_wrap(~Phylum, nrow=2, scales="free_x") + 
  ylab("-log10(p.adj)") + xlab("effect size") + 
  scale_x_continuous(breaks=c(-1,0,1), limits=c(-1.2,1.2)) + 
  theme_cowplot()
  # ggtitle("Effect size vs significance for taxonomic ranks")
species_p
ggsave("figures/Fig_3_volc_plot_species.pdf", width=10, height=5)

wilcox_test_results %>% filter(rank_level=="s" & p.adj < 0.05 & effsize_trans>0) %>% 
  pull(Phylum) %>% table() %>% View()
```
### Comparing two control groups across two major different phyla
```{r}
phy_compare <- taxa_level_summary %>% 
  filter(rank_level=="Phylum" &  !grepl("Cyano",tax_rank)) %>% 
  group_by(tax_rank) %>% 
  wilcox_test(data=., rel_abundance~Experiment)# %>% 
  # adjust_pvalue(method="BH")

phy_compare_eff <- taxa_level_summary %>% 
  filter(rank_level=="Phylum" & grepl("Control",Experiment) & !grepl("Cyano",tax_rank)) %>% 
  group_by(tax_rank) %>% 
  wilcox_effsize(data=., rel_abundance~Experiment)

phy_compare_summary <- taxa_level_summary %>% 
  filter(rank_level=="Phylum" & grepl("Control",Experiment) & !grepl("Cyano",tax_rank)) %>% 
  group_by(tax_rank,Experiment) %>% 
  summarize(mean=mean(rel_abundance))
  
```

## NMDS plot
```{r, echo=FALSE}
u_dist <- distance(ps_rel, method="unifrac")
u_nmds <- ordinate(ps_rel, method="NMDS", distance=u_dist)$points %>% 
  cbind(sample_data(ps_rel)) %>% 
  data.frame() %>% rownames_to_column("SampleName")
```

```{}
#old
u_nmds_plot <- ggplot(u_nmds, aes(x=MDS1, y=MDS2, shape=factor(exp_group))) + 
  geom_point(aes(fill=Experiment, size=timepoint), alpha=0.5) + 
  scale_fill_manual(values=c("Control 1"="white","Filtered stool"="gray","Unfiltered stool"="black",
                              "Control 2"="white", "Co-housed w/ diarrhea"="black")) + 
  # scale_color_manual(values=c("Control 1"="white","Filtered stool"="gray","Unfiltered stool"="black",
  #                             "Control 2"="white", "Co-housed w/ diarrhea"="black")) + 
  scale_shape_manual(values=c("1"=24, "2"=22)) + 
  scale_size_continuous(range=c(2,4)) + 
  guides(size="none", 
         guide_legend(override.aes = list(shape = c(24, 24, 24, 22, 22)))) +
  geom_line(aes(group=mouse_id), alpha=0.5) + 
  xlab("NMDS1") + ylab("NMDS2") + 
  theme_cowplot()
u_nmds_plot
```

```{r}
u_nmds_plot <- ggplot(u_nmds, aes(x=MDS1, y=MDS2, shape=origin_expgroup)) + 
  geom_point(aes(fill=Exposure, size=factor(timepoint)), alpha=0.8) + #size = 3
  scale_shape_manual("Mouse, experiment", 
                     values=c("Jax, gavage"=24, "Stanford, co-house"=22),
                     guide = guide_legend(override.aes = list(size=3))) + 
  scale_fill_manual("Diarrhea exposure", values=c("control"="white","filtered"="gray","unfiltered"="black"), 
                    guide = guide_legend(override.aes = list(shape=(21), fill = c("white","gray","black"), size=3))) + 
  scale_size_manual("Timepoint", values=c("1"=2, "2"=3),
                    guide = guide_legend(override.aes = list(shape=21, fill="white"))) +
  # guides(size="none") +
  geom_line(aes(group=mouse_id), alpha=0.2, size = 0.2) + 
  xlab("NMDS1") + ylab("NMDS2") + 
  scale_x_continuous(breaks=c(seq(-0.4, 0.3, 0.2))) + 
  scale_y_continuous(breaks=c(-0.2,0,0.2), limits = c(-0.25,0.2)) + 
  coord_fixed() + 
  theme_cowplot()
u_nmds_plot
ggsave("figures/Fig_1_nmds.pdf", plot=u_nmds_plot, height=4)
```
## Plotting % reads mapped results
```{r}
read_map_data <- sample_data(ps_rel) %>% data.frame() %>% 
  rownames_to_column("SampleName") %>% 
  select(SampleName, unique_reads, reads_mapped_MGBC, reads_mapped_MGBC_add41) %>% 
  pivot_longer(3:4, names_to="database", values_to="reads_mapped") %>% 
  mutate(perc_reads_mapped=reads_mapped/unique_reads)

read_map_p <- ggplot(read_map_data, aes(x=database, y=perc_reads_mapped)) + 
  geom_boxplot() + 
  # geom_beeswarm(size=1.2) + 
  # facet_wrap(~diarrhea) +
  ylab("Fraction of reads mapped") + 
  # ggtitle("Read recruitment to three reference databases") + 
  geom_signif(comparisons=list(c("reads_mapped_MGBC","reads_mapped_MGBC_add41")),
              step_increase = 0.15) + 
  xlab("Reference database used") + 
  scale_y_continuous(breaks=seq(0,1,0.2), limits=c(0,1.2)) + 
  scale_x_discrete(labels=c("reads_mapped_MGBC"="MGBC", "reads_mapped_MGBC_add41"="MGBC + 41")) + 
  theme_cowplot()
read_map_p
ggsave("figures/Fig_1_read_recruitment.pdf", plot = read_map_p, width=3, height=5)
```
## Plotting novel bins
```{r}
new_bins_metadata <- read_excel("Supplementary Tables.xlsx", sheet="S3_bins_metadata")
write(new_bins_metadata$`Bin Id`, "tables/bin_id_246.txt")

bin_plot <- ggplot(new_bins_metadata, aes(x=mash_dist, y=score, color=novel)) + 
  geom_point(alpha=1, shape=1, size=3) + 
  ylab("Quality score of genome bin") + xlab("ANI distance from nearest MGBC genome") + 
  # ggtitle("ANI vs quality of 246 bins (medium-quality or better)") + 
  scale_color_manual("Novel", values=c('FALSE'="gray",'TRUE'="black"),
                     labels=c("FALSE"="no", "TRUE"="yes")) + 
  geom_vline(xintercept = 0.05, color="black", linetype="dashed") + 
  theme_cowplot()
bin_plot
ggsave("figures/Fig_1_novel_bins.pdf", plot=bin_plot, width=5, height=5)
```

# FIGURE 2
Creating shared species distance matrix
```{r}
t <- otu_table(ps) %>% data.frame() %>% 
       rownames_to_column("genome") %>% 
  pivot_longer(2:27, names_to="SampleName", values_to="rel_ab") %>% 
  filter(rel_ab>0)

ps_pres_abs <- ps_rel %>% transform_sample_counts(function(OTU) ifelse(OTU>0,1,0) ) %>% 
  otu_table() %>% as.matrix()

shared_species <- function(x, y) { 
  set_x <- x[x==1] %>% rownames()
  set_y <- y[y==1] %>% rownames()
  return(as.numeric(length(intersect(set_x, set_y))))
}

shared_species_j <- function(x, y) { 
  set_x <- x[x==1] %>% rownames()
  set_y <- y[y==1] %>% rownames()
  i <- as.numeric(length(intersect(set_x, set_y)))
  u <- as.numeric(length(union(set_x, set_y)))
  return(i/u)
}

shared_dist <- sapply(1:ncol(ps_pres_abs) ,function(y) sapply(1:ncol(ps_pres_abs),function(x) shared_species(ps_pres_abs[,x],ps_pres_abs[,y]))) # WORKED!
rownames(shared_dist) <- colnames(ps_pres_abs)
colnames(shared_dist) <- rownames(shared_dist)
# View(shared_dist)

shared_dist_long <- data.frame(shared_dist) %>% 
  rownames_to_column("SampleName") %>% 
  pivot_longer(2:27, names_to="SampleName2", values_to="num_shared_species") %>% 
  arrange(SampleName, SampleName2) %>% 
  distinct()

jaccard_dist <- distance(ps_rel, method = "jaccard", binary = TRUE)
# jdist2 <- sapply(1:ncol(ps_pres_abs) ,function(y) sapply(1:ncol(ps_pres_abs),function(x) shared_species_j(ps_pres_abs[,x],ps_pres_abs[,y])))
# rownames(jdist2) <- colnames(ps_pres_abs)
# colnames(jdist2) <- rownames(jdist2)

jaccard_similarity <- 1-as.matrix(jaccard_dist) %>% data.frame()#
jaccard_similarity_long <- jaccard_similarity %>% 
  rownames_to_column("SampleName") %>% 
  pivot_longer(2:27, names_to="SampleName2", values_to="jaccard_similarity") %>% 
  arrange(SampleName, SampleName2)
```

Creating shared strains distance matrix - per mouse
```{r}
instrain_genomewide <- read_excel("Supplementary Tables.xlsx", 
                                  sheet="S5_instrain_shared") %>%
  mutate(name1=gsub("__on__MGBC_1094_reps_41_add.sorted.bam","",name1),
         name2=gsub("__on__MGBC_1094_reps_41_add.sorted.bam","",name2)) %>% 
  filter(grepl("FF",name1) & grepl("FF",name2,)) %>% 
  dplyr::rename(SampleName=name1, SampleName2=name2) %>% 
  filter(popANI >= 0.99999)

shared_strain_count <- instrain_genomewide %>% select(SampleName, SampleName2) %>% 
  # The next line is needed in order to create a symmetrical distance matrix later on
  rbind(data.frame(SampleName=instrain_genomewide$SampleName2, SampleName2=instrain_genomewide$SampleName) %>% filter(SampleName!=SampleName2)) %>% 
  arrange(SampleName, SampleName2) %>% 
  summarize.(shared_strains=n(), .by=c("SampleName","SampleName2"))

strains_combined <- shared_dist_long %>% 
  left_join(shared_strain_count) %>% 
  mutate.(shared_strains = ifelse(is.na(shared_strains),0,shared_strains),
          perc_strains_shared_of_max = shared_strains / max(shared_strains), 
          perc_strains_diff_of_max = 1 - perc_strains_shared_of_max, 
          # SampleName=factor(SampleName, levels=sample_order),
          # SampleName2=factor(SampleName2, levels=sample_order),
          .by=SampleName) %>% 
  left_join(jaccard_similarity_long) %>% 
  mutate(SampleName=factor(SampleName, levels=sample_order),
         SampleName2=factor(SampleName2, levels=sample_order),
         perc_strains_shared_of_max=ifelse(perc_strains_shared_of_max==0,NA,perc_strains_shared_of_max),
         ) %>% 
  arrange(SampleName, SampleName2)

shared_strains_dist <- strains_combined %>% 
  select(SampleName, SampleName2, perc_strains_shared_of_max) %>% 
  pivot_wider(names_from=SampleName2, values_from=perc_strains_shared_of_max) %>% 
  column_to_rownames("SampleName")

strains_vs_genomes_shared <- strains_combined %>% 
  filter(SampleName==SampleName2)

strains_vs_genomes_shared %>% 
  summarize(shared_species_mean=mean(num_shared_species),
             shared_species_sd=sd(num_shared_species),
             shared_strains_mean=mean(shared_strains),
             shared_strains_sd=sd(shared_strains),
             )
```


```{r}
strains_shared_summary <- strains_combined %>% 
  left_join.(md_metadata %>% select(SampleName, Experiment)) %>% 
  left_join.(md_metadata %>% select(SampleName, Experiment) %>% 
               rename.(SampleName2=SampleName, Experiment2=Experiment))


strains_shared_summary%>% 
  filter(SampleName!=SampleName2) %>% 
  summarize(shared_species_mean=mean(num_shared_species),
             shared_species_sd=sd(num_shared_species),
             shared_strains_mean=mean(shared_strains),
             shared_strains_sd=sd(shared_strains),
             )
```
```{r}
strains_shared_summary%>% 
  filter(SampleName!=SampleName2 & Experiment==Experiment2) %>% 
  summarize(shared_species_mean=mean(num_shared_species),
             shared_species_sd=sd(num_shared_species),
             shared_strains_mean=mean(shared_strains),
             shared_strains_sd=sd(shared_strains),
             )
```

Making a heatmap:
```{r, fig.width=5}
ggplot(strains_combined, aes(x=SampleName, y=reorder(SampleName2, desc(SampleName2)))) + 
  geom_tile(aes(fill=perc_strains_diff_of_max)) + 
  xlab("SampleName") + ylab("SampleName") + 
  coord_fixed() + 
  theme(axis.text.x = element_text(angle = 90))
```

```{r}
cluster_samples <- hclust(dist(shared_strains_dist), method = "complete")
```

```{r}

heatmap_metadata <- sample_data(ps_rel) %>% data.frame() %>% select(Exposure, origin_expgroup) %>% 
  rename("Mouse, experiment"=origin_expgroup)

anno_colors <- list(
  Exposure = c("control"="white","filtered"="gray","unfiltered"="black"),
  `Mouse, experiment` = c("Jax, gavage"="#8c9f19", "Stanford, co-house" = "#9f2c19")
)

pheatmap(mat = as.matrix(shared_strains_dist), 
         # cluster_rows = cluster_samples,
         # cluster_cols = cluster_samples,
         cluster_rows=F, cluster_cols=F, gaps_row=c(6,12,18,22), gaps_col=c(6,12,18,22),
         border_color=NA,
         # cutree_rows = 4,
         # cutree_cols = 4,
         annotation_row = heatmap_metadata, annotation_col = heatmap_metadata,
         show_colnames=F,
         show_rownames=F,
         # treeheight_row = 0,
         # treeheight_col=20,
         annotation_colors = anno_colors,
         color=blues(100),
         # color = colorRampPalette(rev(brewer.pal(n=7,name="YlGnBu")))(100),
         # filename = "figures/paper/Fig_2_strain_heatmap_unclustered_newcolor.pdf",
         cellwidth = 10,
         cellheight = 10,
         # na_col = "#283A80",
         legend_breaks = seq(0,1,0.2),
         # clustering_dista
)
```
strain-sharing analysis
```{r}
control2 <- sample_data(ps_rel) %>% data.frame() %>% filter(Experiment=="Control 2") %>% rownames()
unfiltered <- sample_data(ps_rel) %>% data.frame() %>% filter(Experiment=="Unfiltered stool") %>% rownames()
chd <- sample_data(ps_rel) %>% data.frame() %>% filter(Experiment=="Co-housed w/ diarrhea") %>% rownames()

stanford_strains <- instrain_genomewide %>% 
  filter((SampleName %in% control2 & SampleName2 %in% chd) | (SampleName2 %in% control2 & SampleName %in% chd)) %>% 
  pull(genome) %>% sort() %>% unique

stanford_strains_in_gavage <- instrain_genomewide %>% 
  filter((SampleName %in% control2 & SampleName2 %in% unfiltered) | (SampleName2 %in% control2 & SampleName %in% unfiltered)) %>% 
  pull(genome) %>% sort() %>% unique()

diarrhea_strains_in_gavage <- instrain_genomewide %>% 
  filter((SampleName %in% chd & SampleName2 %in% unfiltered) | (SampleName2 %in% chd & SampleName %in% unfiltered)) %>% 
  pull(genome) %>% sort() %>% unique()

diarrhea_candidates <- setdiff(diarrhea_strains_in_gavage, union(stanford_strains_in_gavage, stanford_strains))
```

Statistical results for diarrhea candidates:
```{r}
gtdb_df_mod <- read_excel("Supplementary Tables.xlsx", sheet="S4_species_rep_tax") %>% 
  separate.(classification, sep=";", into=c("k","p","c","o","f","g","s"))

diarrhea_cand_joined <- gtdb_df_mod %>% filter.(user_genome %in% diarrhea_candidates) %>% select(user_genome, s) %>% 
  dplyr::rename(tax_rank=s) %>% 
  left_join.(wilcox_test_results)
```




Gives us 3 Muribaculaceae, 6 Alistipes, and the 2 predicted pathogens.
How many are shared species?
```{r}
# otu_table(ps_rel)[diarrhea_candidates,] %>% t() %>% View()

ps_rel_diarrhea <- ps %>% 
  filter_taxa(function(x){sum(x>0)>3}, prune=T) %>% 
  transform_sample_counts(function(x) x+1) %>% 
  transform_sample_counts(function(OTU) (OTU/sum(OTU))*100 ) %>%
  subset_taxa(rownames(otu_table(physeq)) %in% diarrhea_candidates) %>% 
  transform_sample_counts(log10) %>% 
  transform_sample_counts(function(OTU) ifelse(OTU<(-3.9),NA,OTU) )

# p <- plot_heatmap(ps_rel_diarrhea)

bin_order <- c("MGBC165708","MGBC163040","MGBC128991","MGBC104261","MGBC151524","MGBC111863","MGBC116472","MGBC110745","MGBC143807","METABAT215_SAMESTATE_PRE_SCAFFOLDS_1500_METASPADES__diarrhea_F405_F406_F409_F410_FC732_FC733_MC728.160","MGBC105482")

diarrhea_otu_matrix <- as.matrix(otu_table(ps_rel_diarrhea))[bin_order,sample_order]

diarrhea_row_md <- tax_table(ps_rel_diarrhea) %>% data.frame() %>% select(Family)
diarrhea_row_labels <- tax_table(ps_rel_diarrhea)[bin_order,"Species"] %>% 
  data.frame() %>% 
  mutate(Species=gsub("s__","",Species)) %>% 
  as.matrix()

# cluster_diarrhea_taxa <- hclust(dist(diarrhea_otu_matrix), method = "average")

pheatmap(mat = diarrhea_otu_matrix,
          cluster_rows=F, gaps_row=c(3,9),
          cluster_cols=F, gaps_col=c(6,12,18,22),
          clustering_distance_rows = "euclidean",
          show_colnames=F,
          show_rownames=T,
          border_color=NA,
          annotation_row = diarrhea_row_md,
          annotation_col = heatmap_metadata,
          # color=blues(100),
          color = colorRampPalette(rev(brewer.pal(n=7,name="YlGnBu")))(100),
          annotation_colors = anno_colors,
          na_col = "#283A80",
          labels_row = diarrhea_row_labels,
         filename = "figures/Fig_2_abundance.pdf",
         cellwidth = 10,
         cellheight = 10,
 )
```
FC732 had intermittent diarrhea on day 1, severe diarrhea on day 25. Not many things had massive blooms except our two candidate pathogens.

Add:
Volcano plot?
Phylogenomic tree for clostridia genus
Spore forming (presence across tree)
Toxin production genes (presence across tree)
Genomic context of toxin genes?
PCR primers